frontend/src/components/ExplorerCard/ExplorerCard.tsx

```diff
@@ -0,0 +1,329 @@
+import {
+	DeleteOutlined,
+	DownOutlined,
+	MoreOutlined,
+	SaveOutlined,
+	ShareAltOutlined,
+} from '@ant-design/icons';
+import {
+	Button,
+	Card,
+	Col,
+	Dropdown,
+	MenuProps,
+	Popover,
+	Row,
+	Space,
+	Typography,
+} from 'antd';
+import TextToolTip from 'components/TextToolTip';
+import {
+	queryParamNamesMap,
+	querySearchParams,
+} from 'constants/queryBuilderQueryNames';
+import { useGetSearchQueryParam } from 'hooks/queryBuilder/useGetSearchQueryParam';
+import { useQueryBuilder } from 'hooks/queryBuilder/useQueryBuilder';
+import { useDeleteView } from 'hooks/saveViews/useDeleteView';
+import { useGetAllViews } from 'hooks/saveViews/useGetAllViews';
+import { useUpdateView } from 'hooks/saveViews/useUpdateView';
+import useErrorNotification from 'hooks/useErrorNotification';
+import { useNotifications } from 'hooks/useNotifications';
+import { mapCompositeQueryFromQuery } from 'lib/newQueryBuilder/queryBuilderMappers/mapCompositeQueryFromQuery';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useCopyToClipboard } from 'react-use';
+import { ViewProps } from 'types/api/saveViews/types';
+
+import { ExploreHeaderToolTip } from './constants';
+import MenuItemGenerator from './MenuItemGenerator';
+import SaveViewWithName from './SaveViewWithName';
+import {
+	DropDownOverlay,
+	ExplorerCardHeadContainer,
+	OffSetCol,
+} from './styles';
+import { ExplorerCardProps } from './types';
+import {
+	deleteViewHandler,
+	getViewDetailsUsingViewKey,
+	isQueryUpdatedInView,
+	updateQueryHandler,
+} from './utils';
+
+function ExplorerCard({
+	sourcepage,
+	children,
+	currentPanelType,
+}: ExplorerCardProps): JSX.Element {
+	const [isOpen, setIsOpen] = useState<boolean>(false);
+	const [copyUrl, setCopyUrl] = useCopyToClipboard();
+	const [isQueryUpdated, setIsQueryUpdated] = useState<boolean>(false);
+	const { notifications } = useNotifications();
+
+	const onCopyUrlHandler = (): void => {
+		setCopyUrl(window.location.href);
```

Comment:
use location from hook try not to use window object

Reply:
To copy the full URL, we have to use `window.location.href` as `useLocation` will only give the `pathname` and `search` strings.